modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/GridDhtPartitionDemander.java

```diff
@@ -1641,17 +1723,41 @@ private synchronized void partitionDone(UUID nodeId, int p, boolean own) {
             assert rmvd : "Partition already done [grp=" + grp.cacheOrGroupName() + ", fromNode=" + nodeId +
                 ", part=" + p + ", left=" + parts + "]";
 
-                if (rmvd)
-                    partitionsLeft.decrementAndGet();
+            if (rmvd)
+                partitionsLeft.decrementAndGet();
 
             if (parts.isEmpty()) {
                 int remainingRoutines = remaining.size() - 1;
 
-                U.log(log, "Completed " + ((remainingRoutines == 0 ? "(final) " : "") +
+                int fullParts = fullRebalancingParts.get(nodeId).size();
+                int histParts = histRebalancingParts.get(nodeId).size();
+
+                long fullEntries = fullReceivedKeys.get(nodeId).sum();
+                long histEntries = histReceivedKeys.get(nodeId).sum();
+
+                long fullBytes = fullReceivedBytes.get(nodeId).sum();
+                long histBytes = histReceivedBytes.get(nodeId).sum();
+
+                long duration = System.currentTimeMillis() - startTime;
+                long durationSec = Math.max(1, TimeUnit.MILLISECONDS.toSeconds(duration));
+
+                U.log(log, "Completed " + (remainingRoutines == 0 ? "(final) " : "") +
                     "rebalancing [grp=" + grp.cacheOrGroupName() +
                     ", supplier=" + nodeId +
+                    ", partitions=" + (fullParts + histParts) +
+                    ", entries=" + (fullEntries + histEntries) +
+                    ", duration=" + duration + "ms" +
```

Comment:
Rebalancing duration can be rather big and not human readable in millis.
Can you implement utility method durationToString to humanReadableByteCount ?
It should output human readable durations, like 1h20m14s. Check for example
`PeriodFormat.getDefault().print(duration.toPeriod())`